src/main/webapp/app/shared/metis/metis-conversation.service.ts

```diff
@@ -184,20 +184,10 @@ export class MetisConversationService implements OnDestroy {
         );
     };
 
-    public setUpConversationService = (courseId: number): Observable<never> => {
-        if (!courseId) {
-            throw new Error('CourseId is not set. The service cannot be initialized.');
-        }
-        this._courseId = courseId;
-        this.setIsLoading(true);
-        return this.courseManagementService.findOneForDashboard(courseId).pipe(
-            map((res: HttpResponse<Course>) => {
-                return res.body;
-            }),
-            switchMap((course: Course) => {
-                this._course = course;
-                return this.conversationService.getConversationsOfUser(this._course.id ?? 0);
-            }),
+    public setUpConversationService = (course: Course): Observable<never> => {
+        this._courseId = course.id!;
+        this._course = course;
```

Comment:
I think you need to have `this.setIsLoading(true);` here. Or is this removal intended?

Reply:
It was intentional since the api call is removed here, so loading is not needed.